docs/BillOfMaterials.md

```diff
@@ -0,0 +1,170 @@
+Bill of Materials
+=================
+
+The bill of materials (bom) is a list of artifacts produced by the build.
+
+## Generating a new bom
+
+A new bill of materials is generated on each build and placed in `artifacts/bom.$(Version).xml`.
+
+In addition, this bill of materials can be produced without building the repo by running `build.ps1 -bom`.
+
+## Configuration data
+
+:x: The bom SHOULD NOT contain data that is for configuration. Configuration data is any kind of parameter that a consumer needs to find the artifact.
+
+Examples of ephemeral data:
+
+ - File paths
+ - Download links
+ - Passwords
+
+## Format
+
+The bill of materials is an XML file. Here is a minimal example of a bom.
+
+```xml
+<Build>
+
+    <Artifacts>
+        <Artifact Id="Microsoft.Extensions.Configuration.Json.2.0.0.nupkg" Type="NuGetPackage" />
+    </Artifacts>
+
+    <Artifacts Category="noship">
+        <Artifact Id="Experimental.1.0.0-alpha.123.nupkg" Type="NuGetPackage" />
+    </Artifacts>
+
+    <Dependencies>
+        <Link Source="Microsoft.Extensions.Configuration.Json.2.0.0-rtm-123.nupkg" Target="Newtonsoft.Json.10.0.1.nupkg" />
+        <Link Id="Experimental.1.0.0-alpha.123.nupkg" Target="Microsoft.Extensions.Configuration.Json.2.0.0-rtm-123.nupkg" />
+    </Dependencies>
+
+</Build>
+```
+
+### Elements
+
+#### `<Build>`
+
+Describes the build output. The root element of the document.
+
+Optional attributes:
+  - `Id` - a unique identifier describing this build.
+  - `Created` - a timestamp of when the build was created.
+
+#### `<Artifacts>`
+
+Describes a group of artifacts
+
+Optional attributes:
+  - `Category` - an arbitrary category name.
+
+#### `<Artifact>`
+
+Describes a file produced by the build.
+
+Required attributes:
+  - `Id` - the file name of the artifact. Id MUST BE unique per each bill of materials.
+  - `Type` - an arbitrary string describing the semantic type of the artifact.
+
+Optional attributes:
+    - Others - any other attributes on the element are considered metadata values about the artifact.
+
+#### `<Dependencies>`
+
+Describes artifact dependencies.
+
+#### `<Link>`
+
+Describes a dependency between two artifacts.
```

Comment:
This file isn't meant to be human readable is it? I'm imagining that the list of dependencies can grow pretty large. Also, is this a full closure of dependencies or is it the top level dependencies?

Reply:
Super-human readable only. The file is really meant to be consumed by other tools.

Just top-level dependencies.